MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/JoinProcessor/EntityId.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\SearchCriteria\CollectionProcessor\JoinProcessor;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\JoinProcessor\CustomJoinInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Data\Collection\AbstractDb;
+
+/**
+ * Class EntityType
+ * Used to join table of the collection based on the filter selection.
+ * @package Magento\MediaGalleryUi\Model\SearchCriteria\CollectionProcessor\JoinProcessor
+ */
+class EntityId implements CustomJoinInterface
+{
+    private const MEDIA_CONTENT_ASSET_TABLE_NAME = 'media_content_asset';
+
+    private const CMS_PAGE_TABLE_NAME = 'cms_page';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * @param ResourceConnection $connection
+     */
+    public function __construct(ResourceConnection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function apply(AbstractDb $collection): bool
+    {
+        $collection->getSelect()->joinLeft(
```

Comment:
This join is going to be added in https://github.com/magento/adobe-stock-integration/pull/1531 no need to join the same table twice